src/path.c

```diff
@@ -236,6 +236,152 @@ void twin_path_arc(twin_path_t *path,
+{
+    twin_fixed_t dot = twin_fixed_mul(ux, vx) + twin_fixed_mul(uy, vy);
+
+    twin_fixed_t ua =
+        twin_fixed_sqrt(twin_fixed_mul(ux, ux) + twin_fixed_mul(uy, uy));
+    twin_fixed_t va =
+        twin_fixed_sqrt(twin_fixed_mul(vx, vx) + twin_fixed_mul(vy, vy));
+
+    /* cos(theta) = (u ⋅ v) / (|u| * |v|) */
+    twin_fixed_t cos_theta = twin_fixed_div(dot, twin_fixed_mul(ua, va));
+    twin_fixed_t cross = twin_fixed_mul(ux, vy) - twin_fixed_mul(uy, vx);
+    twin_angle_t angle = twin_acos(cos_theta);
+    return (cross < 0) ? -angle : angle;
+}
+
+typedef struct {
+    twin_fixed_t cx, cy;
+    twin_angle_t start, extent;
+} twin_ellipse_param_t;
+
+static twin_ellipse_param_t get_center_parameters(twin_fixed_t x1,
+                                                  twin_fixed_t y1,
+                                                  twin_fixed_t x2,
+                                                  twin_fixed_t y2,
+                                                  bool fa,
+                                                  bool fs,
+                                                  twin_fixed_t rx,
+                                                  twin_fixed_t ry,
+                                                  twin_fixed_t phi)
+{
+    twin_fixed_t sin_phi = twin_sin(phi);
+    twin_fixed_t cos_phi = twin_cos(phi);
+
+    /* Simplify through translation/rotation */
+    twin_fixed_t x =
+        twin_fixed_mul(cos_phi, twin_fixed_mul(x1 - x2, TWIN_FIXED_HALF)) +
+        twin_fixed_mul(sin_phi, twin_fixed_mul(y1 - y2, TWIN_FIXED_HALF));
+
+    twin_fixed_t y =
+        twin_fixed_mul(-sin_phi, twin_fixed_mul(x1 - x2, TWIN_FIXED_HALF)) +
+        twin_fixed_mul(cos_phi, twin_fixed_mul(y1 - y2, TWIN_FIXED_HALF));
+
+    twin_fixed_t px = twin_fixed_mul(x, x);
+    twin_fixed_t py = twin_fixed_mul(y, y);
+    twin_fixed_t prx = twin_fixed_mul(rx, rx);
+    twin_fixed_t pry = twin_fixed_mul(ry, ry);
+    /* Correct out-of-range radii */
+    twin_fixed_t L = twin_fixed_div(px, prx) + twin_fixed_div(py, pry);
+
+    if (L > TWIN_FIXED_ONE) {
+        twin_fixed_t sqrt_L = twin_fixed_sqrt(L);
+        rx = twin_fixed_mul(sqrt_L, twin_fixed_abs(rx));
+        ry = twin_fixed_mul(sqrt_L, twin_fixed_abs(ry));
+    } else {
+        rx = twin_fixed_abs(rx);
+        ry = twin_fixed_abs(ry);
+    }
+
+    /* Compute center */
+    twin_fixed_t sign = (fa != fs) ? -1 : 1;
+    double px_d = twin_fixed_to_double(px);
+    double py_d = twin_fixed_to_double(py);
+    double prx_d = twin_fixed_to_double(prx);
+    double pry_d = twin_fixed_to_double(pry);
+
+    twin_fixed_t A =
+        twin_double_to_fixed(pry_d / (py_d + (pry_d / prx_d) * px_d));
+    twin_fixed_t B =
+        twin_double_to_fixed(prx_d / (prx_d + (px_d / py_d) * pry_d));
+    twin_fixed_t C =
+        twin_double_to_fixed(pry_d / (pry_d + (py_d / px_d) * prx_d));
+
+    twin_fixed_t pM = A - B - C;
+    twin_fixed_t M = sign * twin_fixed_sqrt(pM);
```

Comment:
Why it cannot use `twin_fixed_mul` and `twin_fixed_div`?

Reply:
I am currently drawing a TinyVG image, and I found that the input value for an ellipse might be too large and could cause overflow (I have tried `twin_dfixed`(Q23.8), but it would still overflow).
A example is
```
large_arc : 0 sweep : 1 rx 3423.000000 ry 3423.000000 cur 409.750000 28.250000 tar 401.375000 37.125000
```

## [comic.tvg](https://tinyvg.tech/img/comic.png)

## original (double)
<img width="640" alt="image" src="https://github.com/user-attachments/assets/1401b271-7148-4ced-bdf4-f233568f0698">

## overflow (Q16.16)
<img width="639" alt="image" src="https://github.com/user-attachments/assets/840de2ad-f0f6-48e4-b41f-03d2c79ef018">

## [chart.tvg](https://tinyvg.tech/img/chart.png)
ellipse parameter:
```
large_arc : 0 sweep : 1 rx 3423.000000 ry 3423.000000 cur 409.750000 28.250000 tar 401.375000 37.125000
```
## original (double)
<img width="642" alt="image" src="https://github.com/user-attachments/assets/258f6d71-fd23-4452-b7cd-934782fcb8a4">

## overflow (Q16.16)
<img width="640" alt="image" src="https://github.com/user-attachments/assets/aa6ff4fa-ef3f-49f7-8a35-637aaa915c57">

Reply:
> I am currently drawing a TinyVG image, and I found that the input value for an ellipse might be too large and could cause overflow (I have tried twin_dfixed(Q23.8), but it would still overflow).

How about fixed-point type in Q31.32 format?
Type Name: `twin_xfixed_t` where "x" could denote "extended precision"

| Aspect  | Q47.16 | Q31.32  |
|---------|--------|--------|
| Range   | ±140 × 10¹²    | ±2.1 × 10⁹  |
| Precision  | Moderate (≈ 0.000015)   | High (≈ 2.33 × 10⁻¹⁰)    |
| Implementation   | Easier to manage scaling and rounding      | Requires careful scaling and rounding   |